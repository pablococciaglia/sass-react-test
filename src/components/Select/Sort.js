import React from 'react'

export const Sort = ({setSelect}) => {


	function tuggleshow() {
		let sort = document.getElementsByClassName("selector__option");
		if (sort[0].style.display === "none") {
			sort[0].style.setProperty('display', 'block');
		} else {
			sort[0].style.setProperty('display', 'none');
		}
		
	}

	const handleOptionChange = changeEvent => {
		setSelect (changeEvent.target.value);
		tuggleshow(); 
	};

	return (
		<>
		<div className="header__margin">
		
			<div>
			<h1 className="header__title">LOTTOLAND GAMES</h1>
			</div>
			<div onClick={tuggleshow} className="selector__frame selector__select">
				<p><span>1 &nbsp;</span>SORT GAMES</p>
			</div>
			<form id="selector__option" className="selector__option selector__select" onClick={tuggleshow} onChange={handleOptionChange} >
				<p><span>2 &nbsp;</span>  SORT GAMES</p>
				<input type="radio" id="test1" name="inputradio" value="name" /><label for="test1">Name (A-Z)</label><br/>
				<input type="radio" id="test2" name="inputradio" value="stakemin" /><label for="test2">Stake (max to min)</label><br/>
				<input type="radio" id="test3" name="inputradio" value="stakemax" /><label for="test3">Stake (min to max)</label><br/>
			</form>
		</div>
		<hr/>
		</>
	)
}
