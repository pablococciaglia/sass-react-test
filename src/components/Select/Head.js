import React, { useState } from 'react'
import { Sort } from './Sort'

export const Head = ({setSelect, select}) => {
    const [optionlist, setoptionlist] = useState(false)
    
    const handleOptionlist = ()=>{
        if (optionlist===false){
            setoptionlist(!optionlist)
        }else{
            setTimeout(() => {
                setoptionlist(!optionlist)
            }, 250);
        }
    }
    
    return (
       <>
            <div className="header__margin">
            
                <div>
                <h1 className="header__title">LOTTOLAND GAMES</h1>
                </div>
                <div onClick={handleOptionlist} className="selector__frame">
                    <p><span>1</span>SORT GAMES</p>
                </div>
                {optionlist && <Sort setSelect={setSelect} select={select} handleOptionlist={handleOptionlist} />}
                
            </div>
            <hr/>
		</>
    )
}
