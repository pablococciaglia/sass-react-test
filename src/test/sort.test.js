import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';
import { Sort } from '../components/Select/Sort';

const wrapper = shallow(<Sort />);

describe('test on Sort.js', () => {
     
    test('create a snapshot of the component', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show the sort (its a "form" element) options on click on the div (SORT GAMES)', () => {
        wrapper.find('div').at(1).simulate('click');
        expect((wrapper.find('form')).exists()).toBe(true);
    })

});
 