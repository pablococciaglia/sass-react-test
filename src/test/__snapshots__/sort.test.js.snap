// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on Sort.js create a snapshot of the component 1`] = `
<Fragment>
  <div
    className="header__margin"
  >
    <h1
      className="header__title"
    >
      LOTTOLAND GAMES
    </h1>
    <div
      className="selector__frame selector__select"
      onClick={[Function]}
    >
      <p>
        <span>
          1 Â  
        </span>
        SORT GAMES
      </p>
    </div>
  </div>
  <hr />
</Fragment>
`;
